ARMmbed/mbedtls
# Fix printf missed issues

## Description
Fix for a couple of issues that got through the CI from the previous CI Patch.

1. Fix printf format issues in a couple of programs
2. Lower the -wformat-truncation level as this causes false positive errors in mbedtls_snprintf. The issue is that any buffer which the compiler does not know the length of is treated as size 1 byte, thus anything being put into that buffer is a potential overflow.

This PR resolves #4233 

## Status
**READY**

## Requires Backporting
NO, however problem exists in 2.26 release.

## Migrations
NO

## Todos


## Steps to test or reproduce
Build latest build cmake with MBEDTLS_HAVEGE_C, MBEDTLS_THREADING_C and MBEDTLS_THREADING_PTHREAD enabled in config.h. Build should pass clean.
